recipes/grpc/all/conanfile.py

```diff
@@ -60,6 +60,16 @@ def requirements(self):
         self.requires('abseil/20210324.1')
         self.requires('re2/20210401')
 
+    def build_requirements(self):
+        self.build_requires('protobuf/3.17.1')
+
+        #when cross compiling we need pre compiled grpc plugins for protoc
+        if tools.cross_building(self):
+            if not hasattr(self, "settings_build"):
+                return
+            else:
+                self.build_requires('grpc/{}'.format(self.version))
```

Comment:
We are used to see this conditional in CCI:

```suggestion
        if hasattr(self, "settings_build") and tools.cross_building(self):
            self.build_requires('grpc/{}'.format(self.version))
```

Reply:
we can update this for both test package and conanfile once it passes. which i think it will now

Reply:
seems to be the same effect, just a bit different code style, up to you if you want to change

Reply:
updated in both locations